### DISCUSSION TITLE
Tableview Scroll issue, SDK related

### DISCUSSION THREAD

**Discussion by raybelisle** [appcelerator/titanium_mobile#12736] [Q&A] [answered]
I have a tableview that lazy loads data and works great on 9.2.1GA or less and has for over 5 years. When I compile using 9.3.x+ I get weird behaviour on Android. The app adds the rows as required when you reach the end of the table but then the tableview won't scroll up past the previous row total. 

So, the example is, on app start 10 records are retrieved and placed in the table. Then I scroll down to the bottom of the table and 10 additional rows are added to the table. Now the table has 20 total rows. When I scroll back up, the table will not scroll up beyond record 11. It stops there, but the scroll indicator to the right of the table shows I am only half way up the table.

It's really strange because when I compile with 9.2.1GA it all works as expected. I know there were changes to the tableview/listview handling but has anyone else encountered this? Do you know how/if there is a fix?

**Comment by m1ga** [comment]
did you already try a nightly build of 9.3.3 (https://builds.appcelerator.com/mobile/9_3_X/mobilesdk-9.3.3.v20210405082000-osx.zip) or even a 10.0.0 (https://builds.appcelerator.com/mobile/10_0_X/mobilesdk-10.0.0.v20210421092535-osx.zip). 
There are a lot of TableView/ListView commits since they've moved to RecyclerView in 9.3.0. Perhaps it is already fixed. If not it would be good to digg a bit deeper. Would be nice if you can create a simple test case so we test the same

**Reply by raybelisle** [reply]
So, bad news is 9.3.3 does not fix it. Tried 10, but obviously there are other issues because I am getting errors that I have never seen before so likely there are either some issues or some refactoring I need to do to support 10.x which I don't have time to work out right now. Unfortunately, I don't have a 'simple' test case that can recreate the issue at the moment, but I'll have to see if I can create one that will work.

**Reply by m1ga** [reply]
yeah, 10 introduced  Material themes so you have to change some stuff there if you use a custom theme. 
So it might even be good to have a separated test app.

How do you add new elements? Do you append them?

```javascript
const win = Ti.UI.createWindow({
	backgroundColor: 'white'
});

// initial data
var data = [];
for (var i = 0; i < 50; ++i) {
	var r = Ti.UI.createTableViewRow({
		title: 'row a ' + i,
		color: "#000"
	});
	data.push(r)
}
const table = Ti.UI.createTableView({
	data: data
});

var btn = Ti.UI.createButton({
	title: "add"
})
btn.addEventListener('click', e => {
	for (var i = 0; i < 50; ++i) {
		var r = Ti.UI.createTableViewRow({
			color: "#000",
			title: 'row b ' + i,
		});
		table.appendRow(r)
	}
});

win.add(btn);
win.add(table);
win.open();
```

works fine for me. Is your TableView nested into another scrolling view?

**Reply by raybelisle** [reply]
The hierarchy is Window/View/Tableview, so nothing weird. The table data is being kept in an array and after the updates to the array the tableView data is replaced. The other issue is that is fails about 80-90% of the time, but sometimes it works (no scrolling issues) so definitely something wonky.

Also, it is a fairly complex row layout. 2 Different types both with some differing elements.

**Comment by raybelisle** [comment, marked as answer]
Well, it turns out it was the setting of the array to be the tableview data. Not sure why that is occurring in this case, but changing the code to manipulate the tableView directly seems to have avoided the issue. Going to close for now but it is a bit concerning that this issue got past testing and will not be looked at.
